### PR TITLE
Use Micronaut with airbyte-server-wrapped

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Support conversion of the airbyte-server-wrapped Cloud module to Micronaut

## How
* Remove `@Requires` rule from controllers so they will be used in both OSS and Cloud
* Remove `airbyte-commons-worker` dependency from `airbyte-server` by moving shared code to new `airbyte-commons-converters` module

## Recommended reading order
1. `*ApiController.java`

## Testing

* All tests pass
* Project builds locally
* Tested changes in Cloud locally

**N.B.** This will not be merged until the corresponding Cloud PR has been tested in a dev environment.  At the point of merging, this will temporarily break Cloud if Cloud is updated to that OSS commit or later without the corresponding change being merged on the Cloud side.

### CODE REVIEW

File: airbyte-commons-converters/src/main/java/io/airbyte/converters/CatalogClientConverters.java [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
  */
 
-package io.airbyte.workers.helper;
+package io.airbyte.converters;
```
Comment (benmoriceau) [inline airbyte-commons-converters/src/main/java/io/airbyte/converters/CatalogClientConverters.java:5]: Nit: All the other commons have `commons` in their package names, should we keep this pattern for this new module?

File: airbyte-server/src/main/java/io/airbyte/server/apis/ConnectionApiController.java [hunk 2]
```diff
@@ -31,9 +30,7 @@
 import io.micronaut.security.rules.SecurityRule;
 
 @Controller("/api/v1/connections")
-@Context()
-@Requires(property = "airbyte.deployment-mode",
-          value = "OSS")
+@Context
```
Comment (benmoriceau) [inline airbyte-server/src/main/java/io/airbyte/server/apis/ConnectionApiController.java:33]: Should we remove`@Context` or add it to the other controller as well? I'm not sure that it has any impact on cloud. It was just an annotation I added after noticing slowness on my local test.

File: airbyte-server/src/main/java/io/airbyte/server/apis/NotificationsApiController.java [hunk 3]
```diff
@@ -13,23 +13,20 @@
...
-@Requires(property = "airbyte.deployment-mode",
-          value = "OSS")
+@Controller("/api/v1/notifications")
 @Secured(SecurityRule.IS_AUTHENTICATED)
 public class NotificationsApiController implements NotificationsApi {
 
   public NotificationsApiController() {}
 
-  @Post
+  @Post("/try")
```
Comment (benmoriceau) [inline airbyte-server/src/main/java/io/airbyte/server/apis/NotificationsApiController.java:29]: Should we do the same for the other controllers (like the openApi one)

File: airbyte-server/src/main/java/io/airbyte/server/config/ApplicationBeanFactory.java [hunk 4]
```diff
@@ -109,6 +110,11 @@ public AirbyteProtocolVersionRange airbyteProtocolVersionRange(
     return new AirbyteProtocolVersionRange(new Version(minVersion), new Version(maxVersion));
   }
 
+  @Singleton
```
Comment (benmoriceau) [inline airbyte-server/src/main/java/io/airbyte/server/config/ApplicationBeanFactory.java:113]: Why is it needed in the server? It feel strange to have a need for a client in the server. Is the server doing request to the server?

**Review by benmoriceau — APPROVED** [review]
I have nit comments that shouldn't block merging that.

File: airbyte-commons-converters/src/main/java/io/airbyte/converters/CatalogClientConverters.java (hunk 1 above)
Comment (jdpgrailsdev) [reply airbyte-commons-converters/src/main/java/io/airbyte/converters/CatalogClientConverters.java:5]: Probably.  I will do a rename.

File: airbyte-server/src/main/java/io/airbyte/server/apis/ConnectionApiController.java (hunk 2 above)
Comment (jdpgrailsdev) [reply airbyte-server/src/main/java/io/airbyte/server/apis/ConnectionApiController.java:33]: I think that we should do a pass on this after we get these changes in.  I'm hesitant to remove anything until we get everything working and have a baseline.

File: airbyte-server/src/main/java/io/airbyte/server/config/ApplicationBeanFactory.java (hunk 4 above)
Comment (jdpgrailsdev) [reply airbyte-server/src/main/java/io/airbyte/server/config/ApplicationBeanFactory.java:113]: It is used by the `OAuthHandler`, presumably to fetch OAuth tokens.

**Pull request merged**